### PR TITLE
Only send AttorneyTasks back to judge assign

### LINKED ISSUE #15218: Cancellation of an AttorneyRewriteTask should reactivate JudgeDecisionReviewTask, not JudgeAssignTask
## Description
AttorneyRewriteTasks inherit their behavior from AttorneyTasks. Upon cancelling an AttorneyTask we cancel the associated JudgeDecisionReviewTask and open a new JudgeAssignTask. This is correct behavior for a normal AttorneyTask. However when we have an AttorneyRewriteTask the JudgeDecisionReviewTask is properly working, and we should not cancelled it.

Override the cancel flow behavior in AttorneyRewriteTasks

## Acceptance criteria
- [ ] AttorneyRewriteTasks return control to the JudgeDecisionReviewTasks, instead of cancelling them and creating a JudgeAssignTask
- [ ] Bulk Reassign handles this case properly
- [ ] Update any Bat Team docs

## Background/context/resources
[Example case](https://dsva.slack.com/archives/CHX8FMP28/p1599769654200500)

### PR DESCRIPTION
**Pull request by hschallhorn**
Resolves #15218

### Description
Cancelling ONLY `AttorneyTask`s will send a case back to judge assign.

### Acceptance Criteria
- [ ] When cancelling a task, `AttorneyTask` _subtypes_ return control to their parent, instead of cancelling them and creating a `JudgeAssignTask`
- [ ] When cancelling a task, `AttorneyTasks` cancel their parent `JudgeDecisionReviewTask` and create a `JudgeAssignTask`

### Testing Plan
1. Log in as a dispatch user and send a case back to a judge
1. Log in as the judge and send the case back to an attorney
```ruby
Appeal.find_by_uuid("dc932435-af1b-4f07-8d37-f0a1088597ce").treee
                                               ┌──────────────────────────────────────────────────────────────────────────────┐
Appeal 346 (hearing) ───────────────────────── │ ID   │ STATUS      │ ASGN_BY       │ ASGN_TO       │ UPDATED_AT              │
└── RootTask                                   │ 972  │ on_hold     │               │ Bva           │ 2020-10-19 14:09:00 UTC │
    ├── DistributionTask                       │ 973  │ on_hold     │               │ Bva           │ 2020-10-19 14:09:00 UTC │
    │   └── HearingTask                        │ 974  │ on_hold     │               │ Bva           │ 2020-10-19 14:09:00 UTC │
    │       └── ScheduleHearingTask            │ 975  │ assigned    │               │ Bva           │ 2020-10-19 14:09:00 UTC │
    ├── JudgeDecisionReviewTask                │ 976  │ completed   │ CSS_ID17      │ BVAAWAKEFIELD │ 2020-10-19 14:09:01 UTC │
    │   └── AttorneyTask                       │ 977  │ completed   │ BVAAWAKEFIELD │ BVAABELANGER  │ 2020-10-19 14:09:00 UTC │
    └── BvaDispatchTask                        │ 978  │ on_hold     │               │ BvaDispatch   │ 2020-10-19 14:09:01 UTC │
        └── BvaDispatchTask                    │ 979  │ on_hold     │               │ BVAGWHITE     │ 2020-10-26 19:48:36 UTC │
            └── JudgeDispatchReturnTask        │ 3415 │ on_hold     │ BVAGWHITE     │ BVAAABSHIRE   │ 2020-10-26 20:07:39 UTC │
                └── AttorneyDispatchReturnTask │ 3416 │ in_progress │ BVAAABSHIRE   │ BVAEERDMAN    │ 2020-10-26 20:08:13 UTC │
                                               └──────────────────────────────────────────────────────────────────────────────┘
```
1. Log in as the attorney and cancel the task.
1. Confirm this does not send the case back to judge assign and instead makes the parent task "assigned"
```ruby
Appeal.find_by_uuid("dc932435-af1b-4f07-8d37-f0a1088597ce").treee
                                               ┌────────────────────────────────────────────────────────────────────────────┐
Appeal 346 (hearing) ───────────────────────── │ ID   │ STATUS    │ ASGN_BY       │ ASGN_TO       │ UPDATED_AT              │
└── RootTask                                   │ 972  │ on_hold   │               │ Bva           │ 2020-10-19 14:09:00 UTC │
    ├── DistributionTask                       │ 973  │ on_hold   │               │ Bva           │ 2020-10-19 14:09:00 UTC │
    │   └── HearingTask                        │ 974  │ on_hold   │               │ Bva           │ 2020-10-19 14:09:00 UTC │
    │       └── ScheduleHearingTask            │ 975  │ assigned  │               │ Bva           │ 2020-10-19 14:09:00 UTC │
    ├── JudgeDecisionReviewTask                │ 976  │ completed │ CSS_ID17      │ BVAAWAKEFIELD │ 2020-10-19 14:09:01 UTC │
    │   └── AttorneyTask                       │ 977  │ completed │ BVAAWAKEFIELD │ BVAABELANGER  │ 2020-10-19 14:09:00 UTC │
    └── BvaDispatchTask                        │ 978  │ on_hold   │               │ BvaDispatch   │ 2020-10-19 14:09:01 UTC │
        └── BvaDispatchTask                    │ 979  │ on_hold   │               │ BVAGWHITE     │ 2020-10-26 19:48:36 UTC │
            └── JudgeDispatchReturnTask        │ 3415 │ assigned  │ BVAGWHITE     │ BVAAABSHIRE   │ 2020-10-26 20:09:25 UTC │
                └── AttorneyDispatchReturnTask │ 3416 │ cancelled │ BVAAABSHIRE   │ BVAEERDMAN    │ 2020-10-26 20:09:25 UTC │
                                               └────────────────────────────────────────────────────────────────────────────┘
```

### CODE REVIEW

File: app/models/tasks/attorney_task.rb [hunk 1]
```diff
@@ -63,13 +63,13 @@ def send_back_to_judge_assign!(params = {})
...
-    update_params_will_cancel_task?(params) ? send_back_to_judge_assign!(params) : super(params, user)
+    update_params_will_cancel_attorney_task?(params) ? send_back_to_judge_assign!(params) : super(params, user)
   end
 
   private
 
-  def update_params_will_cancel_task?(params)
-    params[:status].eql?(Constants.TASK_STATUSES.cancelled)
+  def update_params_will_cancel_attorney_task?(params)
+    type == AttorneyTask.name && params[:status].eql?(Constants.TASK_STATUSES.cancelled)
```
Comment (hschallhorn) [inline app/models/tasks/attorney_task.rb:72]: The change! Only AttyTasks go back to judge assign, not subclasses

File: spec/factories/task.rb [hunk 2]
```diff
@@ -416,7 +416,7 @@ def self.find_first_task_or_create(appeal, task_type)
         parent { create(:ama_judge_decision_review_task, appeal: appeal) }
       end
 
-      factory :ama_judge_dispatch_return_to_attorney_task, class: AttorneyDispatchReturnTask do
+      factory :ama_attorney_dispatch_return_task, class: AttorneyDispatchReturnTask do
```
Comment (hschallhorn) [inline spec/factories/task.rb:419]: sneaky rename

File: spec/models/tasks/attorney_task_spec.rb [hunk 3]
```diff
@@ -169,8 +172,10 @@
     end
 
     context "when cancellation params are provided" do
+      let(:params) { { status: Constants.TASK_STATUSES.cancelled } }
+
       it "calls send_back_to_judge_assign!" do
-        tasks = attorney_task.update_from_params({ status: Constants.TASK_STATUSES.cancelled }, attorney)
+        tasks = subject
```
Comment (hschallhorn) [inline spec/models/tasks/attorney_task_spec.rb:178]: General refactoring

**Review by lomky — APPROVED** [review]
:tada: :shipit:

**Pull request merged**